zerver/models.py

```diff
@@ -1679,3 +1685,27 @@ class CustomProfileFieldValue(models.Model):
 
     class Meta(object):
         unique_together = ('user_profile', 'field')
+
+class Service(models.Model):
+    name = models.CharField(max_length=UserProfile.MAX_NAME_LENGTH, unique=True) # type: Text
+    # owner of service
+    user_profile = models.ForeignKey(UserProfile) # type: UserProfile
+    # address of the third party site
+    base_url = models.TextField() # type: Text
+    # used for authentication to third party site
+    token = models.TextField() # type: Text
+    # the interface used to send data to third party site
+    interface = models.TextField(default=None) # type: Text
```

Comment:
Should this be an enum-type field (see Recipient for an example of how this is usually done)?

It seems like we'll probably just have a fixed like of like 3 interfaces, not an endless list, right?

Otherwise, looks great!